docs/data/toolpad/getting-started/why-toolpad.md

```diff
@@ -7,41 +7,65 @@
+### Slower delivery speed
 
-### A lot of boilerplate
+Every organization wants to achieve fast delivery but in order to accommodate various use cases, you are pushed to choose flexible pro-code solutions.
 
+This flexibility comes at the cost of more **boilerplate** code.
 To connect backend data to the browser and set up data-display components, a substantial amount of boilerplate HTML, CSS, and JavaScript glue code is often required.
+As the use cases repeat themselves, you create abstractions but their maintenance burden quickly grows.
+
+Wouldn't it be great if you could use a tool that abstracts the repetitive parts of creating internal tools?
+
+### No direct manipulation of objects
+
+By the definition of a pro-code tool, you write code in an IDE and the UI renders in the browser, a different window.
+
+You can't **directly manipulate** the parameters, queries, etc. in the browser. It makes it harder to make changes on the application.
+
+Wouldn't it be great if it was possible to directly manipulate the app created in the browser, manipulating the data the closest possible to the source of truth?
 
-### Lower-level abstraction
+### Require specialized knowledge
 
-Creating internal tools demands knowledge of a front-end framework. While this aspect might be intriguing to some, most developers of internal tools often consider managing the lower-level APIs, HTML, and CSS as an additional burden.
+Creating internal tools with pro-code requires knowledge of front-end technologies.
+
+While front-end developers feel comfortable with it, most developers of internal tools are full-stack engineers, people who consider writing low-level CSS and React code a **distraction**.
+
+Wouldn't it be great if you could use a tool that is approachable, not requiring to be a React expert?
 
 ## Trade-offs with low-code tools
 
-There exist many low-code code tools specifically for internal use cases. They offer speed, but come with some trade-offs:
+The key problems most low-code tools have relative to Toolpad:
 
 ### Extensibility
 
-Low-code tools can be hard to extend. When the tool can't handle a particular task by default, integrating your own code seamlessly is often a challenge. This lack of flexibility often results in having to write custom code inside these apps and storing your business logic within their configuration files.
+Low-code tools can be hard to extend. When the tool can't handle a particular task by default, integrating your own code seamlessly is often a challenge.
+
+This lack of flexibility often results in being stuck or having to write custom code inside these apps and storing the business logic within their configuration files.
+
+Would it be great if the tool would use code in your filesystem as the source of truth?
```

Comment:
To me the "local first" argument is about ownership. Existing hosted low-code tools store their configurations opaquely on their end. They end up reinventing version control tools and deployment lifecycle tools, which don't integrate well with your existing stack.